plugins/USBPrinting/AutoDetectBaudJob.py

```diff
@@ -55,32 +55,57 @@ def run(self) -> None:
+                        Logger.logException("w", "Unable to create serial: Unknown exception")
+                        continue
                 else:
                     # We already have a serial connection, just change the baud rate.
                     try:
                         serial.baudrate = baud_rate
-                    except ValueError:
+                    except:
+                        Logger.logException("w", "Setting baud rate: Exception")
                         continue
                 sleep(wait_bootloader)  # Ensure that we are not talking to the boot loader. 1.5 seconds seems to be the magic number
                 successful_responses = 0
 
-                serial.write(b"\n")  # Ensure we clear out previous responses
-                serial.write(b"M105\n")
+                try:
+                    serial.write(b"\n")  # Ensure we clear out previous responses
+                    serial.write(b"M105\n")
+                except:
+                    Logger.logException("w", "serial.write: Exception")
+                    continue
 
                 start_timeout_time = time()
                 timeout_time = time() + wait_response_timeout
+                mytime = time()
 
-                while timeout_time > time():
-                    line = serial.readline()
-                    if b"ok" in line and b"T:" in line:
-                        successful_responses += 1
-                        if successful_responses >= 1:
-                            self.setResult(baud_rate)
-                            Logger.log("d", "Detected baud rate {baud_rate} on serial {serial} on retry {retry} with after {time_elapsed:0.2f} seconds.".format(
-                                serial = self._serial_port, baud_rate = baud_rate, retry = retry, time_elapsed = time() - start_timeout_time))
-                            serial.close() # close serial port so it can be opened by the USBPrinterOutputDevice
-                            return
+# Using serial.readline will not work here due to all the noise received during baudrate mismatch
+# Replaced with state machinery which will work
 
-                    serial.write(b"M105\n")
+                while timeout_time > time():
+                    if mytime != time():
+                        serial.write(b"M105\n")
+                        mytime = time()
+                    mychar = serial.read()
+                    if mychar == b"\n":
+                        mychar = serial.read()
+                        if mychar == b"o":
+                            mychar = serial.read()
+                            if mychar == b"k":
+                                mychar = serial.read()
+                                if mychar == b" ":
+                                    mychar = serial.read()
+                                    if mychar == b"T":
+                                        mychar = serial.read()
+                                        if mychar == b":":
```

Comment:
These nested `if`s are just bad style. Could you try this with a loop? Something like:

```
expected = "\nok T:"
for expected_char in expected:
    mychar = serial.read()
    if mychar != expected_char:
        break
else:
    successful_responses += 1
    ...
```

(This code was written from the top of my head without testing. Please test first.)

Reply:
Not "bad style" at all - standard readable optimized interpreter code. Not so readable but even more optimized would be using &&.
Anyway had some time to clean up a lot of mess and *really* bad code aka zombie code or dead-beef.
Now the serial parameters are actually set, buffer is cleared before detection and the boot log from printer included in the log. (one of mine was missing an initial filament parameter)
Cheers
Karsten